src/mdns/mdns_mdnssd.cpp

```diff
@@ -361,16 +428,20 @@ void PublisherMDnsSd::HandleServiceRegisterResult(DNSServiceRef         aService
 {
     OTBR_UNUSED_VARIABLE(aDomain);
 
-    otbrError       error = DNSErrorToOtbrError(aError);
-    std::string     originalInstanceName;
-    ServiceIterator service = FindPublishedService(aServiceRef);
+    std::string            type = aType;
+    std::string            originalInstanceName;
+    ServiceRegistrationPtr serviceReg = FindServiceRegistration(aServiceRef);
+
+    VerifyOrExit(serviceReg != nullptr);
 
-    VerifyOrExit(service != mServices.end());
+    if (type.back() == '.')
+    {
+        type.pop_back();
+    }
```

Comment:
I wonder why we didn't need this before?

Reply:
Because we were using `IsServiceTypeEqual` to handle this case but now we are removing this method in this PR (I didn't remove it before your comment) as it's that useful if we can handle the mdnssd case explicitly.